File: Iv Layhort/Java_Layhort/src/Task_2/Exercise_2_FinalPrice.java
package Task_2;
import java.util.*;
public class Exercise_2_FinalPrice {
    public static void main(String[] args) {
        double productPrize, commissionRate, discountRate;
        Scanner input = new Scanner(System.in);
        System.out.println("Enter the product price, commission rate and discount rate respectively");
        productPrize = input.nextDouble();
        commissionRate = input.nextDouble();
        discountRate=input.nextDouble();

        System.out.println("The final prize is "+finalPrize(productPrize,commissionRate,discountRate));
    }

    private static double finalPrize(double productPrize,double commissionRate,double discountRate){
        double commissionAmount = (productPrize*commissionRate)/100.00;
        double discountAmount = ((commissionAmount + productPrize) * discountRate)/100.00;
        return productPrize + commissionAmount - discountAmount;
    }
}

File: Iv Layhort/Scala-Layhort/src/Task_2/Exercise_2_FinalPrice.scala
package Task_2
import scala.io.StdIn._

object Exercise_2_FinalPrice {
  def main(args: Array[String]): Unit = {
    println("Enter the product price,commission rate and discount rate respectively :")
    val Array(productPrice,commissionRate,discountRate) = readLine().split(" ").map(_.toInt)
    println(s"The final price is ${calculateFinalPrice(productPrice,commissionRate,discountRate)}")
  }

  val calculateFinalPrice:(Double,Double,Double)=> Double = (productPrice:Double,commissionRate:Double,discountRate:Double)=>{
    val commissionAmount = productPrice * (commissionRate/100.00)
    val discountAmount = ((commissionAmount+productPrice)*discountRate)/100.00

    (productPrice + commissionAmount) - discountAmount
  }
}